### PR TITLE
[CT-1630] Convert column_types test for dbt-bigquery

### PR DESCRIPTION
**Pull request by McKnight-42**
resolves #https://github.com/dbt-labs/dbt-core/issues/6404

### Description

convesion of 056 to new functional test method

todo before merge: revert dev_requirements dbt-core pin

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-bigquery/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-bigquery/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Review by colin-rogers-dbt — APPROVED** [review]

File: tests/functional/adapter/column_types/test_column_types.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import pytest
```
Comment (mikealfare) [inline tests/functional/adapter/column_types/test_column_types.py:1]: Is there a reason to separate this from `test_alter_column_types.py`? The two fixtures (module level variables) are copies of fixtures in the other file. So the only new content is the class with two methods.

I would keep the name `test_column_types.py` as it's more generic, so I guess I'm saying to move the other file's contents here technically.

Comment (McKnight-42) [reply tests/functional/adapter/column_types/test_column_types.py:1]: These are separated for now as it is only currently tested in bigquery and is how it was historically, this does also raise a fine question of how big of a change in functionality a test is doing should they be before they are their own file/test but still grouped together.

Comment (VersusFacit) [reply tests/functional/adapter/column_types/test_column_types.py:1]: I'd suggest just revising this to have the fixtures in the same file, then import them, thereby reducing the duplicate code. Best of both worlds. 

Nit: do that

**Review by VersusFacit — APPROVED** [review]

File: tests/functional/adapter/column_types/test_column_types.py (hunk 1 above)
Comment (McKnight-42) [reply tests/functional/adapter/column_types/test_column_types.py:1]: There are a few very slight differences in the fixtures for each test, one easily fixed by a project_config_update in that we materialize alt model as a table, the second for the schema being a change in the int64 type

**Pull request merged**